py/desitarget/targets.py

```diff
@@ -655,10 +658,13 @@ def finalize(targets, desi_target, bgs_target, mws_target,
     # - create a new unique TARGETID
     targets = rfn.rename_fields(targets,
                                 {'OBJID': 'BRICK_OBJID', 'TYPE': 'MORPHTYPE'})
-    targetid = encode_targetid(objid=targets['BRICK_OBJID'],
-                               brickid=targets['BRICKID'],
-                               release=targets['RELEASE'],
-                               sky=sky)
+    # allow TARGETID to be passed as an input (specifically for the mocks).
+    if targetid is not None:
+        targetid = encode_targetid(objid=targets['BRICK_OBJID'],
+                                   brickid=targets['BRICKID'],
+                                   release=targets['RELEASE'],
+                                   sky=sky)
+    assert ntargets == len(targetid)
 
```

Comment:
...and here are the only places (so far) I've had to touch the nominal (non-mock) target selection code.  These changes should hopefully be non-controversial, but I wanted to explicitly alert you.

Reply:
Shouldn't this be
```
if targetid is None:
```
rather than
```
if targetid is not None:
```
?
In other words, the default behavior is for `targetid` to be `None` but currently, if that default is the case, then the clause:

```
 if targetid is not None:
        targetid = encode_targetid(objid=targets['BRICK_OBJID'],
```
will be skipped and a TARGETID will never be calculated?